salt/grains/core.py

```diff
@@ -2225,7 +2225,10 @@ def _hw_data(osdata):
             if os.path.exists(contents_file):
                 try:
                     with salt.utils.files.fopen(contents_file, 'r') as ifile:
-                        grains[key] = ifile.read().strip()
+                        try:
+                            grains[key] = ifile.read().strip().decode('utf-8')
```

Comment:
This will break on Python 3, since PY3's `str` type has no `.decode()` attribute. Please change this to:

```python
grains[key] = salt.utils.stringutils.to_unicode(ifile.read().strip())
```

This will make the code work on both Python 2 and Python 3.

Reply:
Note that this means you will need to add `salt.utils.stringutils` to the imports at the top of the file.